ngraph_bridge/ngraph_utils.cc

```diff
@@ -463,6 +475,198 @@ bool IsProcessedByNgraphPass(Graph* g) {
+      if (dp->d_type == DT_REG)  // look at all files in directory
+      {
+        files.push_back(std::string(dp->d_name));
+      }
+    }
+
+    closedir(dir);
+  } catch (...) {
+    return errors::Internal("Error inspecting files in directory: " + str_path);
+  }
+  // done inspecting directory's files
+
+  // create event object
+  tensorflow::Event event;
+  tensorflow::Summary::Value* summary_value;
+  tensorflow::Env* env = Env::Default();
+
+  event.set_wall_time(env->NowMicros() / 1e6);
+  event.set_step(step);
+  summary_value = event.mutable_summary()->add_value();
+  summary_value->set_tag(dname + " Compute Time (ms)/" + cluster);
+  summary_value->set_simple_value((float)compute_time);
+  // done creating event object
+
+  if (files.empty())  // create new tf event file
+  {
+    str_path += ("stats" + to_string(file_idx));
+
+    tensorflow::EventsWriter writer(str_path);
+
+    try {
+      writer.WriteEvent(event);
+    } catch (...) {
+      return errors::Internal("Error writing event to new event file");
+    }
+  } else  // append to tf event file in dir
+  {
+    std::string file_path = str_path + files[0];
+
+    // open tf event file at location file_path
+    std::unique_ptr<tensorflow::WritableFile> writable_file;
+
+    env->NewAppendableFile(file_path, &writable_file);
+    tensorflow::io::RecordWriter record_writer(writable_file.get());
+    // done opening tf event file at location file_path
+
+    try {
+      record_writer.WriteRecord(event.SerializeAsString());
+    } catch (...) {
+      return errors::Internal("Error updating event file with new event");
+    }
+  }
+
+  return Status::OK();
+}
+
+Status CreateSummaryFromGraph(tensorflow::Graph* graph,
+                              std::string filename_prefix) {
+  // convert Graph* to GraphDef*
+  tensorflow::GraphDef* gdef = nullptr;
+  graph->ToGraphDef(gdef);
+
+  // create event summary writer
+  tensorflow::EventsWriter writer(filename_prefix);
+  tensorflow::Event event;
+
+  // write graph to event summary
+  try {
+    event.set_graph_def(gdef->SerializeAsString());
+    writer.WriteEvent(event);
+  } catch (...) {
+    return errors::Internal("Error writing graph to event file");
+  }
+
+  return Status::OK();
+}
+
+Status CreateSummaryFromGraphDef(tensorflow::GraphDef* graph_def,
+                                 std::string filename_prefix) {
+  // create event summary writer
+  tensorflow::EventsWriter writer(filename_prefix);
+  tensorflow::Event event;
+
+  // write graph to event summary
+  try {
+    event.set_graph_def(graph_def->SerializeAsString());
+    writer.WriteEvent(event);
+  } catch (...) {
+    return errors::Internal("Error writing graph def to event file");
+  }
+
+  return Status::OK();
+}
+
+void AddSessionNameAttr(int file_idx, std::set<string> nodes, Graph* graph) {
+  for (auto node : graph->op_nodes()) {
+    if (node->type_string() == "NGraphEncapsulate") {
+      node->AddAttr(("_session_name" + to_string(file_idx)),
```

Comment:
Since you are adding an attribute here are some things to take care of:

1. we have a sort-of convention that we begin optional attr names that we add with `_ngraph`. so maybe name it `_ngraph_session_name + ...`
2. Whatever you name it, make sure to filter it out in `ParseNodeAttributes`: https://github.com/tensorflow/ngraph-bridge/blob/d7a0b4a033302a9305ae418e419901f083c343bb/ngraph_bridge/ngraph_encapsulate_impl.cc#L476

Reply:
no changes because "_ngraph" prefix to attribute will send attribute to backend. don't want to do that for "_session_name"